### PR TITLE
Added Set Polygon Material Index node

### PR DESCRIPTION
**Pull request by 3DSinghVFX**
I have added the **Set Polygon Material Index** node which allows to set material index per polygon,
Screenshot from 2020-02-16 17-47-56

Example,
Screenshot from 2020-02-16 17-53-41

### CODE REVIEW

**Comment by OmarEmaraDev** [comment]
Is there a reason why you aren't using `foreach_set` here? I think this node should be similar to the other data set nodes, no need for all of those extra options.

**Comment by 3DSinghVFX** [comment]
Omar, I'm using the `foreach_set` function in this node.

You mean only the _ALL_ option always? This node is similar to _Set Vertex Weight_ node.

**Comment by OmarEmaraDev** [comment]
Yes. *Set Vertex Weight* is a special case because its data is stored per element, not contiguously. So the node should be similar to the other set nodes.

**Comment by 3DSinghVFX** [comment]
Okay, I'll make the changes.

**Comment by 3DSinghVFX** [comment]
I have made the changes.

File: animation_nodes/nodes/mesh/set_polygon_material_index.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+        if object is None: return None
+
+        if object.type != "MESH":
+            self.raiseErrorMessage("Object is not a mesh object.")
+
+        if object.mode != "OBJECT":
+            self.raiseErrorMessage("Object is not in object mode.")
+
+        if self.useMaterialIndexList:
+            if min(materialIndices) < 0:
```
Comment (OmarEmaraDev) [inline animation_nodes/nodes/mesh/set_polygon_material_index.py]: You can use the `containsValueLowerThan(0)` method here.

Comment (3DSinghVFX) [reply animation_nodes/nodes/mesh/set_polygon_material_index.py]: Done.

**Pull request merged**